cchecker_web/upload.py

```diff
@@ -66,6 +77,47 @@ def check_files(files, checker):
     return jsonify(message='Upload successful. Please wait a moment while we process the file...', job_id=job_id, files=successful)
 
 
+def generate_dataset(nc_path, cdl_path):
+    subprocess.call(['ncgen', '-o', nc_path, cdl_path])
+
+
+def check_for_cdl(filename, filepath):
+    # Check for a metadata .cdl file
+    if os.path.splitext(filename)[-1] == '.cdl':
+        nc_file = filename.replace('.cdl', '.nc')
+        nc_path = os.path.join(app.config['UPLOAD_FOLDER'],
+                               encode(nc_file))
+        filepath = update_cdl_dimensions(filepath)
+        generate_dataset(nc_path, filepath)
+        filepath = nc_path
+    return filepath
+
+
+def update_cdl_dimensions(filename):
+    '''
+    Rewrite the cdl file to make all dimensions size 1
+    '''
+    transform = False
+    data = []
+    with open(filename, 'r') as f:
+        while True:
+            line = f.readline()
+            if not line:  # End of file
+                break
+            if 'variables:' in line:
+                transform = False
+            elif transform:  # These lines are dimensions. They should be size 1
+                prefix = line.split('=')[0]
+                line = prefix + '= 1 ;\n'
```

Comment:
I like this instead of using regexes, this is pretty safe and doesn't appear to be susceptible to attack.